src/pages/projects/ProjectPage.tsx

```diff
@@ -1,170 +1,150 @@
 import React, { useState, useEffect } from "react";
-import { grey } from "@mui/material/colors";
 
-import { styled } from "@mui/material/styles";
 import Grid from "@mui/material/Grid";
 import Typography from "@mui/material/Typography";
-import Table from "@mui/material/Table";
-import TableBody from "@mui/material/TableBody";
-import TableCell, { tableCellClasses } from "@mui/material/TableCell";
-import TableContainer from "@mui/material/TableContainer";
-import TableHead from "@mui/material/TableHead";
-import TableRow from "@mui/material/TableRow";
-import Paper from "@mui/material/Paper";
+import CircularProgress from "@mui/material/CircularProgress";
 
 import { ProjectPagedList, ProjectApiAxiosParamCreator, RunAxios } from "icats";
-import JSONContainer  from "../../components/JSONContainer/JSONContainer";
-import { PROJECT_DATA_SAMPLE } from "../../utils/constant";
-
-const StyledTableCell = styled(TableCell)(({ theme }) => ({
-  [`&.${tableCellClasses.head}`]: {
-    backgroundColor: grey[300],
-    color: theme.palette.common.black,
-  },
-  [`&.${tableCellClasses.body}`]: {
-    fontSize: 14,
-  },
-}));
-
-const StyledTableRow = styled(TableRow)(({ theme }) => ({
-  "&:nth-of-type(odd)": {
-    backgroundColor: theme.palette.action.hover,
-  },
-  // hide last border
-  "&:last-child td, &:last-child th": {
-    border: 0,
-  },
-}));
-
-function createData(
-  name: string,
-  calories: number,
-  fat: number,
-  carbs: number,
-  protein: number
-) {
-  return { name, calories, fat, carbs, protein };
-}
-
-interface ITableConstant {
-  displayName: string;
-  jsonKey: string | string[];
-}
-const COLUMN_CONSTANT: ITableConstant[] = [
-  { displayName: "Name", jsonKey: ["name"] },
-  { displayName: "Data", jsonKey: "DATA" },
-  { displayName: "ID", jsonKey: ["id"] },
-  { displayName: "OwnerId", jsonKey: ["ownerId"] },
-  { displayName: "TenantId", jsonKey: ["tenantId"] },
-  { displayName: "TenantName", jsonKey: ["tenantName"] },
-  { displayName: "Active", jsonKey: ["active"] },
-  { displayName: "Information", jsonKey: ["information"] },
-  { displayName: "TimeCreated", jsonKey: ["timeCreated"] },
-  { displayName: "TimeModified", jsonKey: ["timeModified"] },
-  { displayName: "BillingMode", jsonKey: ["billingMode"] },
-  { displayName: "DataSharingEnabled", jsonKey: ["dataSharingEnabled"] },
-  { displayName: "StorageBundleId", jsonKey: ["storageBundle", "id"] },
+import TokenPagination, {
+  ITokenPaginationData,
+  tokenPaginationInit,
+} from "../../container/tokenPagination/TokenPagination";
+import JSONContainer from "../../components/JSONContainer/JSONContainer";
+import { useDialogContext } from "../../container/app/DialogContext";
+import { PROJECT_DATA_SAMPLE } from "../../utils/Constant";
+import CustomTable, { IColumnMapping } from "../../container/table/Table";
+
+const COLUMN_MAPPPING: IColumnMapping[] = [
```

Comment:
I love this mapping, makes things a tad easier than keeping HTML table names + values in separate sections in EJS.